src-ext/extension.ts

```diff
@@ -13,62 +13,106 @@ export function activate(context: vscode.ExtensionContext) {
   let disposable = vscode.commands.registerCommand(
     "stackoverflow-ide.run",
     () => {
+      // console.log(`history: ${context.globalState.get("history")}`);
+      if (context.globalState.get("history") == undefined) {
+        context.globalState.update("history", []);
+      }
+      const searchHistoryList: any = context.globalState.get("history");
```

Comment:
You could have also passed a default value, instead of seeing if it was undefined:
```suggestion
      const searchHistoryList: any = context.globalState.get("history", []);
```